tests/TestCase/Middleware/AuthenticationMiddlewareTest.php

```diff
@@ -160,13 +136,12 @@ public function testApplicationAuthenticationRequestResponse()
 
         $application = $this->getMockBuilder(BaseApplication::class)
```

Comment:
BaseApplication does not implement provider interface. Should it be `TesApp\Application` here?

Reply:
Change this and also tried to mock the interface, I'm always getting `1) Authentication\Test\TestCase\Middleware\AuthenticationMiddlewareTest::testApplicationAuthenticationRequestResponse
InvalidArgumentException: Subject must be an instance of `Authentication\AuthenticationServiceInterface` or `Cake\Core\HttpApplicationInterface`, `Mock_Application_985fb108` given.` this.